src/storage/mod.rs

```diff
@@ -278,17 +278,30 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                 // here.
                 let bypass_locks = TsSet::vec_from_u64s(ctx.take_resolved_locks());
 
-                // Update max_ts and check the in-memory lock table before getting the snapshot
-                async_commit_check_keys(
-                    &concurrency_manager,
-                    iter::once(&key),
-                    start_ts,
-                    ctx.get_isolation_level(),
-                    &bypass_locks,
-                )?;
+                let snap_ctx = if need_check_locks_in_replica_read(&ctx) {
+                    SnapContext {
+                        pb_ctx: &ctx,
+                        read_id: None,
+                        start_ts,
+                        key_ranges: vec![point_key_range(key.clone())],
+                    }
+                } else {
+                    // Update max_ts and check the in-memory lock table before getting the snapshot
+                    async_commit_check_keys(
```

Comment:
Is it possible that replica read is enabled but IsolationLevel is not SI, and this is not the leader? (The same to the following functions)

Reply:
Sorry, I was wrong. I should only check memory locks when isolation level is SI.

Reply:
If this is not leader, snapshot will fail